Assets/Editor/CreatureHydrator.cs

```diff
@@ -0,0 +1,263 @@
+                continue;
+            }
+
+            string[] unitDirectories = Directory.GetDirectories(generaDir);
+
+            foreach (var unitDir in unitDirectories)
+            {
+                string unitName = Path.GetFileName(unitDir);
+                string unitYamlPath = Path.Combine(unitDir, $"{unitName}.yml");
+
+                if (!File.Exists(unitYamlPath))
+                {
+                    Debug.LogError($"YAML file for unit {unitName} not found at {unitYamlPath}");
+                    continue;
+                }
+
+                string yamlContent = File.ReadAllText(unitYamlPath);
+                UnitDefinition unit = DeserializeYaml<UnitDefinition>(yamlContent);
+
+                // Hydrate the creature using the spec and directory
+                HydrateCreature(unit, generaName, unitName, unitDir);
+            }
+        }
+    }
+
+    private static T DeserializeYaml<T>(string yamlContent)
+    {
+        var deserializer = new DeserializerBuilder()
+            .WithNamingConvention(CamelCaseNamingConvention.Instance)
+            .Build();
+
+        return deserializer.Deserialize<T>(yamlContent);
+    }
+
+
+    private static void HydrateCreature(UnitDefinition unit, string generaName, string unitName, string unitDir)
+    {
+        // Create the Prefab
+        GameObject unitPrefab = new GameObject(unitName);
+        SpriteRenderer spriteRenderer = unitPrefab.AddComponent<SpriteRenderer>();  // We'll set the sprite later
+        var animator = unitPrefab.AddComponent<Animator>();
+
+        // Create an Animator Controller within the unit's directory
+        string animatorPath = Path.Combine(unitDir, $"{unitName}Controller.controller");
+        AnimatorController animatorController = AnimatorController.CreateAnimatorControllerAtPath(animatorPath);
+        
+        //Initialize idleSprite so we can set it from the Idle animation sheet
+        Sprite idleSprite = null;
+
+        // Process each animation
+        foreach (var animation in unit.animations)
+        {
+            string animationFolderPath = Path.Combine(unitDir, animation.name);
+            string animationYamlPath = Path.Combine(unitDir, $"{animation.name}/{animation.name}.yml");
+            string animationYamlContent = File.ReadAllText(animationYamlPath);
+            AnimationSpec animationSpec = DeserializeYaml<AnimationSpec>(animationYamlContent);
+
+            string spriteSheetPath = Path.Combine(unitDir, $"{animation.name}/Art/{unitName}_{animation.name}.png");
+            Texture2D texture = AssetDatabase.LoadAssetAtPath<Texture2D>(spriteSheetPath);
+
+            // Slice the sprite sheet
+            Sprite[] sprites = SliceSpriteSheet(texture, animationSpec);
+            // Set the idle sprite as the first frame of the idle animation
+            if (animation.name.Equals("Idle", StringComparison.OrdinalIgnoreCase) && sprites.Length > 0)
```

Comment:
seems a bit smelly in a for loop to grab this one case, I guess in the case where there isn't an idle animation (would that exist?)

Reply:
Oh, it's just so that later we can set this:
```
spriteRenderer.sprite = idleSprite;
```

Which is like the default sprite for the unit (probably should be called that rather than `idleSprite` though)

Reply:
Leaving this for rn tho